### PR TITLE
[horde3d]Add new port

### PR DESCRIPTION
**Pull request by AlvinZhangH**
Related: #3603.

The port horde3d install ok on triplet x86-windows, x64-windows.
Target 'Linux' and "uwp", ARCH "arm" and "arm64" not supported by horde3d currently.
No features need to test.

### CODE REVIEW

File: ports/horde3d/fix-cmakelists.patch [hunk 1]
```diff
@@ -0,0 +1,243 @@
+diff --git a/CMake/Modules/FindGLFW.cmake b/CMake/Modules/FindGLFW.cmake
+deleted file mode 100644
+index 1de274f..0000000
+--- a/CMake/Modules/FindGLFW.cmake
```
Comment (JackBoosY) [inline ports/horde3d/fix-cmakelists.patch]: If you want to delete some files, DO NOT make it into patch. add code to remove it in `portfile.cmake`.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/horde3d/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,28 @@
+vcpkg_fail_port_install(ON_TARGET "uwp")
+vcpkg_fail_port_install(ON_ARCH "arm64")
```
Comment (JackBoosY) [inline ports/horde3d/portfile.cmake]: Should be `vcpkg_fail_port_install(ON_ARCH "arm" "arm64")`

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/horde3d/fix-cmakelists.patch [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
++++ b/Horde3D/Bindings/C++/CMakeLists.txt
+@@ -1,11 +1,17 @@
+ if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
++    install(FILES Horde3D.h Horde3DUtils.h
++            DESTINATION include/horde3d
++            )
+ endif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+ 
+ if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+-        install(FILES Horde3D.h Horde3DUtils.h
```
Comment (JackBoosY) [inline ports/horde3d/fix-cmakelists.patch]: It seems that there is no change except the format. Revert this change.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by PhoebeHui** [comment]
Feel free ask us to reopen this PR if you need it.
